test/named-routes-test.js

```diff
@@ -157,6 +157,61 @@ describe('NamedURLResolver', function() {
+            React.createElement(Route, {name: 'test6', path: '/path(/:param1)/:param2'}),
+
+            React.createElement(Route, {name: 'test7', path: '/path/:param1/(:param2)'}),
+            React.createElement(Route, {name: 'test8', path: '/path/:param1(/:param2)'}),
+
+            React.createElement(Route, {name: 'test9', path: '/path(/:param1/:param2/)'}),
+        ]);
+
+        expect(resolver.resolve("test1", {param: 1})).to.equal("/path/1", "Simple omit case 1");
+        expect(resolver.resolve("test2", {param: 1})).to.equal("/path/1", "Simple omit case 2");
+
+        expect(resolver.resolve("test3", {param2: 2})).to.equal("/path/2", "Middle omit case 1");
+        expect(resolver.resolve("test4", {param2: 2})).to.equal("/path/2", "Middle omit case 2");
+        expect(resolver.resolve("test5", {param2: 2})).to.equal("/path2", "Middle omit case 3");
+        expect(resolver.resolve("test6", {param2: 2})).to.equal("/path/2", "Middle omit case 4");
+
+        expect(resolver.resolve("test7", {param1: 1})).to.equal("/path/1/", "Trailing omit case 1");
+        expect(resolver.resolve("test8", {param1: 1})).to.equal("/path/1", "Trailing omit case 2");
+
+        // if any part of optional sequence is omitted, entire sequence is omitted
+        expect(resolver.resolve("test9", {param1: 1})).to.equal("/path", "Omitted sequence");
+        expect(resolver.resolve("test9", {param1: 1, param2:2})).to.equal("/path/1/2/", "Resolved sequence");
+    });
+
+    it('correctly resolves splat params', function() {
+       resolver.mergeRouteTree([
+            React.createElement(Route, {name: 'test1', path: '/some/*'}),
+            React.createElement(Route, {name: 'test2', path: '/some/*/path'}),
+
+            React.createElement(Route, {name: 'test3', path: '/some/**'}),
+            React.createElement(Route, {name: 'test4', path: '/some/**/path'}),
+            
+            React.createElement(Route, {name: 'test5', path: '/some/*/path/**'})
+        ]);
+
+        expect(resolver.resolve("test1", {splat: 1})).to.equal("/some/1", "Trailing single star");
+        expect(resolver.resolve("test2", {splat: 1})).to.equal("/some/1/path", "Middle single star");
+        expect(resolver.resolve("test1", {splat: "slash/here"})).to.equal("/some/slash_here", "Slash with single star");
+
+        expect(resolver.resolve("test3", {splat: 1})).to.equal("/some/1", "Trailing double star");
+        expect(resolver.resolve("test4", {splat: 1})).to.equal("/some/1/path", "Middle double star");
+        expect(resolver.resolve("test3", {splat: "slash/here"})).to.equal("/some/slash/here", "Slash with double star");
+        
+        expect(resolver.resolve("test5", {splat: "first/splat"})).to.equal("/some/first_splat/path/", "Multiple globs 1");
+        expect(resolver.resolve("test5", {splat: ["first/splat", "second/splat"]})).to.equal("/some/first_splat/path/second/splat", "Multiple globs 2");
```

Comment:
I would add a test where you resolve a route with `id(5)` as a parameter value, or something else that contains parentheses